src/components/form/AutocompleteField.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+  Chip,
+  CircularProgress,
+  makeStyles,
+  TextField,
+  TextFieldProps,
+} from '@material-ui/core';
+import { Autocomplete, AutocompleteProps, Value } from '@material-ui/lab';
+import React from 'react';
+import { Except } from 'type-fest';
+import { FieldConfig, useField, useFieldName } from '.';
+import { getHelperText, showError, useFocusOnEnabled } from './util';
+
+export type AutocompleteFieldProps<
+  T,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
+> = Except<
+  FieldConfig<Value<T, Multiple, DisableClearable, FreeSolo>>,
+  'multiple'
+> & {
+  name?: string;
+  // https://github.com/mui-org/material-ui/issues/19354 -- workaround for setting multiple, freeSolo and disableClearable in a wrapped Autocomplete
+  isMultiple?: boolean;
+  isFreeSolo?: boolean;
+  isDisableClearable?: boolean;
+  hideNoOptionsDiv?: boolean;
+  onOptionSelect?: (options: any[]) => void;
+} & Except<
+    AutocompleteProps<T, Multiple, DisableClearable, FreeSolo>,
+    | 'value'
+    | 'onChange'
+    | 'renderInput'
+    | 'disableClearable'
+    | 'multiple'
+    | 'freeSolo'
+  > &
+  Pick<TextFieldProps, 'helperText' | 'label'>;
+
+const useStyles = makeStyles(() => ({
+  autocompleteInput: {
+    padding: '10px !important',
+  },
+  noOptionsDiv: {
+    display: 'none',
+  },
+}));
+
+export function AutocompleteField<
+  T,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
+>({
+  name: nameProp = 'autocomplete',
+  isMultiple,
+  isFreeSolo,
+  isDisableClearable,
+  hideNoOptionsDiv,
+  defaultValue,
+  disabled: disabledProp,
+  // pass input value to parent component
+  onInput: parentOnInput,
+  onOptionSelect,
+  ...props
+}: AutocompleteFieldProps<T, Multiple, DisableClearable, FreeSolo>) {
+  const name = useFieldName(nameProp);
+  const { input, meta, rest } = useField(name, props);
+  const classes = useStyles();
+  const disabled = disabledProp ?? meta.submitting;
+  const ref = useFocusOnEnabled(meta, disabled);
+  const {
+    required,
+    label,
+    helperText,
+    placeholder,
+    ...autocompleteProps
+  } = rest as any;
+  return (
+    <Autocomplete
+      disabled={disabled}
+      // FF also has multiple and defaultValue
+      multiple={isMultiple}
+      defaultValue={defaultValue}
+      freeSolo={isFreeSolo}
+      disableClearable={isDisableClearable}
+      {...autocompleteProps}
+      classes={{
+        input: classes.autocompleteInput,
+        // hide the "No Options" div when we haven't typed anything yet. Doesn't make sense to show this if we haven't given any input
+        // unless we want to populate an autocomplete with options on initial render any input is given
+        noOptions: hideNoOptionsDiv && classes.noOptionsDiv,
```

Comment:
There's got to be a better way than this. The arrow also changes directions indicating that the popup is open, but it's "not" because have it hidden.